### PR TITLE
Feature: Fix compiler warnings, Refactor and Cleanup

### PR DESCRIPTION
**Pull request by WhoBrokeTheBuild**
This work was originally part of the CMake branch, which I am breaking up into smaller PRs

Add fixes for many compiler errors
Mostly this is just adding `__attribute__((unused))`
Fix some of the formatting broken by clang-format
Replace many `sprintf` with `snprintf`
Rename `wait` to `short_wait` to avoid name conflicts in `UdpEventsTest`
Remove unused test functions for labview

### CODE REVIEW

**Comment by santorofer** [comment]
Everything makes sense, I don't see any obvious issues. Also the formatting make the  changes clear, that helps a lot while reviewing.

**Review by santorofer — DISMISSED** [review]

**Review by mwinkel-dev — DISMISSED** [review]
Most of the changes involve flagging unused parameters to functions and procedures.   The use of `snprintf` to prevent buffer overflow is a good change.   Don't know how many customers still use CAMAC modules, but the `camshr` changes are OK too.

File: examples/demoadc/DemoAdc.c [hunk 1]
```diff
@@ -53,7 +53,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define PI 3.14159265358
 static int totSamples = 65536;
 static double frequency;
-static int pts;
+// static int pts;
```
Comment (zack-vii) [inline examples/demoadc/DemoAdc.c]: I would say delete the line; we have git history to remember

**Review by zack-vii — DISMISSED** [review]

File: examples/demoadc/DemoAdc.c (hunk 1 above)
Comment (joshStillerman) [reply examples/demoadc/DemoAdc.c]: me 2

**Review by mwinkel-dev — APPROVED** [review]
This PR has two commits.   My review and approval of 24-Oct-2024 still applies to the first commit.   The second commit is trivial; it just deletes an extraneous line from `DemoAdc.c` instead of commenting the line.   Thus approving this PR again.

**Pull request merged**